src/Solvers/Random.jl

```diff
@@ -1,8 +1,55 @@
+using OptimizedEinsum: removedsize, ssa_path_cost, ContractionPath
+using OptimizedEinsum.Solvers: greedy_choose_simple!, greedy_choose_thermal!
+using Base: @kwdef
+using Random: seed!
+
 abstract type RandomSolver <: Solver end
 
-struct RandomGreedy <: RandomSolver
-    temperature::Float32
-    rel_temperature::Bool
-    nbranch::Int
-    cost_fn::Core.Function
+@kwdef struct RandomGreedy <: RandomSolver
+    temperature::Float32 = 1.
+    rel_temperature::Bool = true
+    nbranch::Int = 8
+    repeats::Int = 128
+    choose_fn::Function = greedy_choose_thermal!
+    cost_fn::Function = removedsize
+    minimize::Symbol = :flops # :flops or :size
+end
+
+function contractpath(config::RandomGreedy, inputs, output, size)
+    best = Dict{String,Any}("flops" => Inf, "size" => Inf)
+
+    choose_fn = (queue, remaining) -> config.choose_fn(queue, remaining; nbranch=config.nbranch, temperature=config.temperature, rel_temperature=config.rel_temperature)
+
+    trials = [trial_greedy_ssa_path_and_cost(i, inputs, output, size, choose_fn, config.cost_fn) for i in 1:config.repeats]
+
+    # assess the trials
+    for (ssa_path, cost, size) in trials
+
+        if is_better(config.minimize, cost, size, best["flops"], best["size"])
+            best["flops"] = cost
+            best["size"] = size
+            best["ssa_path"] = ssa_path
+        end
+    end
+
+    ContractionPath(best["ssa_path"], inputs, output, size)
+end
+
+function trial_greedy_ssa_path_and_cost(r, inputs, output, size_dict, choose_fn, cost_fn)
+    seed!(r)
```

Comment:
I didn't notice this. Why the `seed`?

Reply:
This `seed` is there to ensure we get a new random path for each trial. From `Solvers.Greedy.greedy_choose_thermal!`:
```julia
    chosen = rand(Categorical(energies))
```

Reply:
But that should be already guaranteed without calling `seed`.

Reply:
You are right, the `seed` is removed in this new commit.